npm/metrics/prometheus-metrics.go

```diff
@@ -55,29 +78,102 @@ const (
 	setHashLabel       = "set_hash"
 )
 
+// Dataplane health metrics
+var (
+	numApplyIPSetFailures         *prometheus.CounterVec
+	numAddPolicyFailures          *prometheus.CounterVec
+	numDeletePolicyFailures       *prometheus.CounterVec
+	numPeriodicPolicyTaskFailures *prometheus.CounterVec
+	numValidatePolicyFailures     *prometheus.CounterVec
+	numValidateIPSetFailures      *prometheus.CounterVec
+)
+
+// Constants for DataplaneHealthMetrics names and descriptions as well as exported labels for Vector metrics
+const (
+	numApplyIPSetFailuresName = "num_apply_ipset_failures"
+	numApplyIPSetFailuresHelp = "The number of times the dataplane failed to apply ipsets"
+
+	numAddPolicyFailuresName = "num_add_policy_failures"
+	numAddPolicyFailuresHelp = "The number of times the dataplane failed to add a network policy"
+
+	numDeletePolicyFailuresName = "num_delete_policy_failures"
+	numDeletePolicyFailuresHelp = "The number of times the dataplane failed to delete a network policy"
+
+	numPeriodicPolicyTaskFailuresName = "num_periodic_policy_task_failures"
+	numPeriodicPolicyTaskFailuresHelp = "The number of times the dataplane failed to run a periodic policy task"
+
+	numValidatePolicyFailuresName = "num_validate_policy_failures"
+	numValidatePolicyFailuresHelp = "The number of times the dataplane failed to validate a network policy"
+
+	numValidateIPSetFailuresName = "num_validate_ipset_failures"
+	numValidateIPSetFailuresHelp = "The number of times the dataplane failed to validate an ipset"
+
+	errorTypeLabel = "error_type"
+)
+
+type RegistryType string
+
+const (
+	CustomerNodeMetrics    RegistryType = "customer-node"
+	CustomerClusterMetrics RegistryType = "customer-cluster"
+	DataplaneHealthMetrics RegistryType = "dataplane-health"
+)
+
 var (
-	nodeLevelRegistry    = prometheus.NewRegistry()
-	clusterLevelRegistry = prometheus.NewRegistry()
-	haveInitialized      = false
+	allRegistries   = []RegistryType{CustomerNodeMetrics, CustomerClusterMetrics, DataplaneHealthMetrics}
+	registries      = make(map[RegistryType]*prometheus.Registry)
+	haveInitialized = false
```

Comment:
Not the biggest fan of globals for storing state. If you get a chance maybe create a `Metric` struct to store state and associate all the functions that you have below to that struct?

Reply:
I like this suggestion, although it would require changes throughout the codebase to use a struct and methods for external calls to the package (if that's what you're suggesting?). Or are you suggesting a Registries struct for internal use?

Reply:
reverting these changes since we will keep the original two registries